.github/workflows/build-nginx-proxy-image.yml

```diff
@@ -0,0 +1,174 @@
+name: Build Nginx Proxy Docker Image
+
+env:
+  IMAGE_NAME: nginx-proxy
+  AWS_REGION: eu-west-1
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Test Pay Nginx Proxy
+    runs-on: ubuntu-latest
+
```

Comment:
Probably worth setting the default shell here on the job so you don't have to repeat it for every step

the other thing is we should definitely set a lower-than-default timeout for the job, because there's quite a lot of waiting for containers to start up, if the containers ever fail to start then the workflow run will take the full default time (which is 360 minutes! WOW!) https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

```yaml
jobs:
  test:
    name: Test Pay Nginx Proxy
    runs-on: ubuntu-latest
    
    timeout-minutes: 15 # We should see how long it actually takes to run the tests and add a few minutes maybe, 15 is just a total guess at something "ok"

    defaults:
        run:
          shell: bash
```